feature.go

```diff
@@ -61,13 +61,33 @@ func NewCollectionFeature(geometries ...*Geometry) *Feature {
 // MarshalJSON converts the feature object into the proper JSON.
 // It will handle the encoding of all the child geometries.
 // Alternately one can call json.Marshal(f) directly for the same result.
-func (f *Feature) MarshalJSON() ([]byte, error) {
-	f.Type = "Feature"
-	if len(f.Properties) == 0 {
-		f.Properties = nil
+func (f Feature) MarshalJSON() ([]byte, error) {
+	type feature struct {
```

Comment:
This is just a copy of `Feature` so you can do `type feature Feature`. Less duplicated code.

Reply:
you're right…